opentimelineio/opentime.py

```diff
@@ -533,29 +576,53 @@ def from_timecode(timecode_str, rate):
 
     :return: (:class:`RationalTime`) Instance for the timecode provided.
     """
+    dropframe = False
+
+    if not isinstance(rate, (int, float)):
```

Comment:
This is true, but isn't it nicer to give an explicit message rather than having to study the traceback?
As suggested in other notes below, I made a rate validation function that at the moment still includes this test. If you don't agree with my arguments for keeping it I'll gladly remove it :)